rest/specs/src/main/filtered-resources/openapi/openapi.yaml

```diff
@@ -0,0 +1,232 @@
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorMessage'
+  parameters:
+    Accept-Version:
+      name: Accept-Version
+      in: header
+      description: >
+        The minimal API version that a client using this API will accept
+        responses from.
+      required: true
+      schema:
+        $ref: '#/components/schemas/Version'
+  headers:
+    ContentVersion:
+      description: >
+        The API version used by the server to produce the REST message. The server will accept
+        messages for any minor versions prior to this one.
+      required: true
+      schema:
+        $ref: '#/components/schemas/Version'
+      examples:
+        major.minor:
+          value: '1.2'
+          summary: A released minor version
+        major.minor.patch:
+          value: '1.0.3'
+          summary: A patched version
+        dev:
+          value: '1.2.0-SNAPSHOT'
+          summary: A development version
+  schemas:
+    Product:
+      description: The JSON description of the Product being ingested.
+      type: object
+      properties:
+
+        productInfo:
+          $ref: '#/components/schemas/ProductInfo'
+        file:
+          $ref: '#/components/schemas/File'
+      required:
+        - productInfo
+        - file
+    ProductInfo:
+        properties:
+          title:
+            type: string
+            minLength: 1
+            description: The title of the Product.
+            example: A Picture
+          filename:
+            type: string
+            minLength: 1
+            description: The name of the source file.
+            example: product.jpeg
+          bytes:
+            type: integer
+            description: The size of the file payload.
+            format: int32
+            minimum: 0
+            exclusiveMinimum: true
+            example: 1024
+          mimeType:
+            type: string
+            description: The MIME type of the file being sent.
+            example: image/jpeg
+        required:
+          - filename
+          - bytes
+          - mimeType
+    File:
+      description: The actual file attachment that is sent in the request.
+      type: string
+      format: binary
+    Version:
+      description: Version number.
+      type: string
+      minLength: 3
+      pattern: '^([1-9]([0-9]+)?)\.(0|([1-9]([0-9]+)?))(\.(0|([1-9]([0-9]+)?))(-.*)?)?$'
+      example: 1.2, 2.4.3, 0.2.5-SNAPSHOT
+    ErrorMessage:
+      description: Error message information.
+      type: object
+      required:
+        - status
+        - message
+      properties:
+        status:
+          type: integer
+          format: int32
+          description: The corresponding HTTP status code.
+          example: 200
+        code:
+          type: integer
+          format: int32
+          description: >
+            An optional more specific code corresponding to the error (specific codes need to be
```

Comment:
❓For this API, should we update the description and provide sub-codes for cases such as when the product size doesn't match the size attribute, or when the product `mimeType` is not supported or is invalid? @paouelle?

Reply:
If you look at my latest spec for the discovery api, I updated this description to indicate that code should be documented as part of each operation. Potentially in the error message description itself. I also documented some codes for the version incompatibility error as a starting point. I decided to go with a convention of having the codes defined as `xxxyyy`, where xxx is the HTTP status code for the code category and `yyy` starts at `001` and increments sequentially and describes a different code. The hope is that we can make sure not to collide across protocol and reuse common codes such that they always mean the same thing. This will be useful for common interceptors like the version interceptor and the authentication one.